truss/remote/baseten/api.py

```diff
@@ -566,3 +567,27 @@ def get_all_secrets(self) -> Any:
 
         secrets_info = resp.json()
         return secrets_info
+
+    def upsert_training_project(self, training_project):
+        headers = self._auth_token.header()
+        resp = requests.post(
+            f"{self._rest_api_url}/v1/training-projects",
+            headers=headers,
+            json={"training_project": training_project.model_dump()},
+        )
+        if not resp.ok:
+            resp.raise_for_status()
+
+        return resp.json()
+
+    def create_training_job(self, project_id: str, job):
```

Comment:
Here (and in the remote wrapper) I ran into a circular dep issue because `truss.base` [ends up](https://github.com/basetenlabs/truss/blob/670b6ffabaa019de898a84fd89fe41f34cc49205/truss-train/truss_train/definitions.py?plain=1#L4) depending on the api. I'm sure we can disentangle this in a followup, but I don't think it needs to block for now